### DISCUSSION TITLE
How to control Audiogridder with a Midi Controller?

### DISCUSSION THREAD

**Discussion by Loupap** [apohl79/audiogridder#1352] [Q&A]
Hello Everyone,

Does anyone have any idea on how to control the audiogridder plugin with a midi controller?

Thanks in advance.

**Comment by oferfuxake** [comment]
I know your post is pretty old by now, but I'll chime in just in case you're still working on this.

I'm using a control surface with my DAW that allows you to control plugin parameters, and when I open the MIDI settings menu on the AG plugin (not the server) I can select "connect automatically" or "connect all" or something like that (not in front of my DAW at the moment, sorry.) The plugin automatically recognizes how many MIDI controls there are available on the control surface and maps that many to the controller. There may or may not be enough knobs or sliders on the controller to cover all the parameters, so it just maps parameters until it runs out of available knobs / sliders. In my case, it knows there are 16 controls so it maps the first 16 parameters. You can change which ones are mapped to the controller in the AG plugin settings menu. So in your case, I'd make sure your MIDI controller is recognized by / visible to your DAW and the AG plugin should be able to see it too.

**Reply by bluetardis** [reply]
One step I had to do / note is increase the 16 channels allocated as some of my channel style plugins need around 40-50.  There is a configuration file you can edit to do this (just stop everything first).  Its in the Client Side notes.

**Reply by Loupap** [reply]
Oxerfuxake and Bluetardis,

Thanks for these replies.

Bluetardis where do I find the client side notes?

Thanks,
Loupap

**Reply by Loupap** [reply]
Also I looked for the "MIDI settings menu on the AG plugin" I can't find it?

Where do I access this through my DAW (Cubase)?

**Reply by bluetardis** [reply]
> where do I find the client side notes
The Client side notes are [here](https://audiogridder.com/plugin/) 


[Server](https://audiogridder.com/server/) is here if you need it.

**Reply by oferfuxake** [reply]
@bluetardis - thanks for pointing us in the right direction! When I get back in front of my AG machines I'm going to be going through the AG configuration file - I knew you could make adjustments not available in the GUI, but I missed a couple of important configurable items that I definitely need to attend to. Configuration file is the key! Great to know. Thank you.

**Reply by Loupap** [reply]
Oberfuxake and Bluetardis,

Thanks for all the info on this it really helps.

I want to edit the configuaration file but I can't seem to find the file on my mac.

I am using the file provided on the plugin page of the website.....but where and how do I edit this file on my mac.

Thanks in advance.

**Reply by oferfuxake** [reply]
@Loupap - I unfortunately have no Mac experience that would be helpful in finding the file. I can tell you that if you do find it you can just open it in a text editor, make changes, and save the file. Super simple. It might be wise to rename the original and keep it around in case you end up getting any unwanted behaviors after editing.

I also want to point out an error in my earlier post - the system doesn't appear to check how many midi controllers there are available. I was wrong about that. In the config file you can set a maximum number of midi control parameters to be mapped. I believe it's actually referred to as automation parameters but these are the same as midi control parameters as far as AG is concerned. I have one plugin with a ton of controls, so I've set my max to something like 40 parameters. (In this case, the channel strip in a very old but great SSL Duende firewire box - yes, AG _can_ work perfectly with the right hardware DSP unit!) If I use a plugin with fewer available controls, it just maps whatever is there. It's very tidy, with the exception of possibly having to assign things like "parameter #1 should control such-and-such plugin knob" etc. since it doesn't know what order you're going to prefer.

**Reply by bluetardis** [reply]
> I want to edit the configuaration file but I can't seem to find the file on my mac.

You are looking for a hidden  folder that lives in your user home folder- sometimes referred to as ~/ 
This will be something like the following however it will have your username.
`/Users/bluetardis/.audiogridder`


The folder is called `.audiogridder` however it is hidden by the Operating System as it has a "." as its first character.  Once you change to that directory you can see the `.audiogridderserver.cfg` file.

I am not sure of your familiarity or tools so the process I will give you two ways to find the file.  
_Close AudioGridder Server before making any changes or bad things will happen._

### Using Finder

- `Command-Tab` to the Smiley Finder Application 
- Press `Command-N` to get a new finder window
- You need to be in your user home directory so either press `Command-Shift-H` or Choose Go/Home from the menu at the top.
- To see hidden files and folders you need to press  `Command-Shift-.` (that last one is a dot/period/whatever).
- Open the .audiogridder folder.
- You can edit the audiogridderserved.cfg with your favourite code editor. 
- If you don't have a simple code editor I would suggest looking up the "CotEditor" app on the apple App Store.  It's free and easy to use for stuff like this.  You don't want to save this with special formatting/characters that most word processors will insert.

### Using Terminal 

- Run Terminal (this is is /Applications/Utilities)
- This will start in the user home directory eg /Users/YourUserName
- You can verify this by typing in `pwd` and pressing Enter.  It will show your directory
- If you type in "`ls -a`" this will list ALL files.
- The .audiogridder folder will be near the top so
- cd .audiogridder/
- and then you can edit this with nano or whatever you are comfortable with.

**Reply by Loupap** [reply]
@bluetardis thank you so much for these detailed steps. I am not a computer person or have any idea on how to program or write code. 

I will probably use the Terminal method. When you talk about editing in the last line and using "nano or whatever you are comfortable with", I guess that means I am not editing in Terminal?

All I want to do is increase the number of available automations from 16 to much higher as I have plugins that have more than 16 controls.

Once again thanks for all your help on this.

Lou

**Reply by bluetardis** [reply]
Nano is a text editor that is built in to terminal.
To use it you would type in 
`nano nameOfFileToEdit.txt`

You can also make a backup first
`cp ./nameoffile.txt ./nameoffile.txt.bkp` or whatever you want to call it.
